include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -0,0 +1,539 @@
+#include <unordered_set>  // for unordered_set
+#include <vector>         // for vector
+
+#include "libsemigroups/present.hpp"  // for Presentation
+#include "libsemigroups/types.hpp"    // for relation_type
+
+namespace libsemigroups {
+  enum class author : uint64_t {
+    Machine    = 0,
+    Aizenstat  = 1,
+    Burnside   = 2,
+    Carmichael = 4,
+    Coxeter    = 8,
+    Easdown    = 16,
+    East       = 32,
+    FitzGerald = 64,
+    Godelle    = 128,
+    Guralnick  = 256,
+    Iwahori    = 512,
+    Kantor     = 1024,
+    Kassabov   = 2048,
+    Lubotzky   = 4096,
+    Miller     = 8192,
+    Moore      = 16384,
+    Moser      = 32768,
+    Sutov      = 65536
+  };
+
+  inline author operator+(author auth1, author auth2) {
+    return static_cast<author>(static_cast<uint64_t>(auth1)
+                               + static_cast<uint64_t>(auth2));
+  }
+
+  std::vector<relation_type> rook_monoid(size_t l, int q);
+
+  // Construct a presentation for the Renner common type B monoid.
+  //
+  // Returns a vector of relations defining the Renner common type B monoid with
+  // parameters `l` and `q`.
+  //
+  // \param l the degree
+  // \param q the parameter `q`
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \noexcept
+  std::vector<relation_type> renner_common_type_B_monoid(size_t l, int q);
+
+  std::vector<relation_type> RennerTypeBMonoid(size_t l, int q);
+
+  // Construct a presentation for the Renner type B monoid.
+  //
+  // Returns a vector of relations defining the Renner type B monoid with
+  // parameters `l` and `q`.
+  //
+  // The argument `val` determines the specific presentation
+  // which is returned. The options are:
+  // * `author::Godelle` (described
+  // [here](https://doi.org/10.48550/arXiv.0904.0926), and is possibly
+  // incorrect)
+
+  // \param l the parameter `l`
+  // \param q the parameter `q`
+  // \param val the author value
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \throws LibsemigroupsException if the author value is not
+  // `author::Godelle`.
+  std::vector<relation_type> renner_type_B_monoid(size_t l, int q, author val);
+
+  // Construct a presentation for the Renner common type D monoid.
+  //
+  // Returns a vector of relations defining the Renner common type D monoid with
+  // parameters `l` and `q`.
+  //
+  // \param l the parameter `l`
+  // \param q the parameter `q`
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \noexcept
+  std::vector<relation_type> renner_common_type_D_monoid(size_t l, int q);
+
+  // Construct a presentation for the Renner type D monoid.
+  //
+  // Returns a vector of relations defining the Renner type D monoid with
+  // parameters `l` and `q`.
+  //
+  // The argument `val` determines the specific presentation
+  // which is returned. The options are:
+  // * `author::Godelle` (described
+  // in [10.48550/arXiv.0904.0926](https://doi.org/10.48550/arXiv.0904.0926),
```

Comment:
You might consider using `[10.48550/arXiv.0904.0926][]` and collecting the links at the end of this doc like:

[10.48550/arXiv.0904.0926]: https://doi.org/10.48550/arXiv.0904.0926

Reply:
[10.48550/arXiv.0904.0926]: https://whatever.xyz